### ISSUE TITLE
another case of odd visual with enrichment

### ISSUE THREAD

**Issue by pkra** [zorkow/speech-rule-engine#459]
This might be more of a MathJax issue but I thought I'd start here. 

Somewhat minmal example: ` \|\hat{f}\|_{X_j}, ` which leads to 

![image](https://user-images.githubusercontent.com/426686/107862676-8868b500-6e4e-11eb-9676-e9c7d35437e3.png)

Affects both on 3.1.1 (on mathjax.org) and 3.2.0-beta as well as on mathjax-src's develop (since I thought this might be fixed in https://github.com/mathjax/MathJax-src/pull/583 but didn't appear to be for me).

**Comment by zorkow** [comment]
This is primarily poor use of LaTeX. 
`\|` produces an operator (see https://latexref.xyz/Math-symbols.html). For use as metric fences one should 
Consequently, MathJax correctly produces an ordinal element rather than an open/close pair using Unicode `&#x2225;`.
SRE similarly interprets this as a parallel operator. 

There is however a MathJax issues, as in the enriched version the second `\parallel` is then interpreted as stretchy, which it should not.

PS: Even, "fixing" this in SRE by interpreting '\|' as a fence, does not fix the MathJax issue. 
I supposed one could interpret parallel as a fence. I am honestly a bit reluctant to do that at the moment until I see more examples, where `&#x2225;` is used as a fence.

**Comment by pkra** [comment]
> `\|` produces an operator (see https://latexref.xyz/Math-symbols.html). For use as metric fences one should

This sentence was cut off.

> There is however a MathJax issues, as in the enriched version the second `\parallel` is then interpreted as stretchy, which it should not.

That's my main concern in this issue.

> PS: Even, "fixing" this in SRE by interpreting '|' as a fence, does not fix the MathJax issue.

But both sides would stetch?

> I supposed one could interpret parallel as a fence. I am honestly a bit reluctant to do that at the moment until I see more examples, where `&#x2225;` is used as a fence.

A rough check says I can give you maybe 10,000 examples across 100 papers.

**Comment by dpvc** [comment]
I haven't been able to reproduce the problem myself.  I get both operators showing `stretchy="false"` even with enrichment.  What am I doing wrong?

Also, MathJax produces U+2225 for the stretchy delimiter as well as for `\|` when used alone.  It would probably be appropriate to use U+2016 for stretchy delimiters and U+2225 for stand-alone `\|`.  The problem is that people will use `\|` without `\left...\right`, as in Peter's example.  Could enrichment substitute U+2016 for U+2225 when it is used in that way?

**Comment by pkra** [comment]
> I haven't been able to reproduce the problem myself. I get both operators showing stretchy="false" even with enrichment. What am I doing wrong?

It reproduces for me on mathjax.org#demo (with the accessibility extension) and using mathjax-src with the stix2 branch (and with develop merged into that branch).

**Comment by dpvc** [comment]
Sorry, I had left off the comma.  D'oh!  Now I see it.

**Comment by zorkow** [comment]
> > `\|` produces an operator (see https://latexref.xyz/Math-symbols.html). For use as metric fences one should
> 
> This sentence was cut off.
> 

I assume I wanted to say "use `\lvert`, `\rvert`, etc.


> 
> > PS: Even, "fixing" this in SRE by interpreting '|' as a fence, does not fix the MathJax issue.
> 
> But both sides would stetch?
> 

No the issue remains the same.

> > I supposed one could interpret parallel as a fence. I am honestly a bit reluctant to do that at the moment until I see more examples, where `&#x2225;` is used as a fence.
> 
> A rough check says I can give you maybe 10,000 examples across 100 papers.

Sounds like it's a common abuse of notation then. I've added it.

**Comment by dpvc** [comment]
OK, I figures out what is going on, and have made a [PR](https://github.com/mathjax/MathJax-src/pull/600) in Mathjax-src to fix it.  The PR includes an explanation of the problem, which turned out not really to be an error in either MathJax or SRE, but an interaction between the two.

**Comment by pkra** [comment]
I think this can be closed as resolved.
